.github/workflows/lint.yml

```diff
@@ -30,4 +30,4 @@ jobs:
           bundle install
       - name: Run Steep
         run: |
-          bundle exec steep check
+          bundle exec steep check --severity-level=error
```

Comment:
It was failing this job because of a warning:

```ruby
lib/datadog/appsec/waf.rb:415:10: [warning] The branch is unreachable because the condition is exhaustive
│ Diagnostic ID: Ruby::ElseOnExhaustiveCase
│
└           ruby_to_object(''.freeze)
            ~~~~~~~~~~~~~~~~~~~~~~~~~
```

In this case, the warning is not 100% true since we have specs that exercise that branch.

Reply:
Let me work things out. 

We have a test that passes an `Object.new` that exercises the else branch. 

https://github.com/DataDog/libddwaf-rb/blob/4d3248f073b20cede98b305c38b04725beafa28c/spec/datadog/appsec/waf_spec.rb#L518-L523